flake.nix

```diff
@@ -80,7 +80,8 @@
             configuration = { ... }: {
               imports = modules
                 ++ [{ programs.home-manager.path = toString ./.; }];
-              nixpkgs = { inherit (pkgs) config overlays; };
+              nixpkgs =
+                nixpkgs.lib.mkDefault { inherit (pkgs) config overlays; };
```

Comment:
Maybe push this down a bit? `nixpkgs.config = mkDefault pkgs.config; nixpkgs.overlays = mkDefault pkgs.overlays;`

Reply:
`config` and `overlays` are the only options under `nixpkgs`
this change does nothing

Reply:
It would allow users to change `nixpkgs.overlays` while still having `config` inherited from `pkgs`, for example.

Reply:
> It would allow users to change `nixpkgs.overlays` while still having `config` inherited from `pkgs`, or vice versa.

```nix
{
  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
    home-manager = {
      url = "github:gerg-l/home-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };
  outputs = {
    nixpkgs,
    home-manager,
    ...
  }: {
    homeConfigurations."gerg" = home-manager.lib.homeManagerConfiguration {
      pkgs = import nixpkgs {
        system = "x86_64-linux";
        config.allowUnfree = true;
      };
      modules = [
        ({pkgs, ...}: {
          nixpkgs.overlays = [
            (
              _: prev: {
                foo = prev.hello;
              }
            )
          ];
          home = {
            packages = [
              pkgs.hello-unfree
            ];
            username = "gerg";
            homeDirectory = "/home/gerg";
            stateVersion = "23.05";
          };
          programs.home-manager.enable = true;
        })
      ];
    };
  };
}
```

Reply:
Oh, of course, `nixpkgs` is not a submodule option so `pushDownProperties` does it for you. If it was, though, my comment would apply, so I don't like relying on this (you never know, it might become a submodule in the future).

Reply:
Code should be written based on present state, not based on what might possibly happen at some point in the future. Over-engineering is less maintainable and doesn't save as much effort as you might assume. If there really needs to be a change, let it be handled at that time.

Reply:
This isn't really about over-engineering, it's about being explicit rather than implicit. There's no need to rely on the "present state" when we could just say what we want explicitly.

Reply:
This is insane in my opinion but whatever, my next PR will fix all this